parsec/core/fs/workspacefs/workspacefile.py

```diff
@@ -87,7 +87,8 @@ async def ainit(self):
                 _, self._fd = await self._transactions.file_create(self._path, open=True)
             else:
                 try:
-                    _, self._fd = await self._transactions.file_open(self._path)
+                    mode = "rw" if self.writable() else "r"
```

Comment:
`b` mode is not taken into account here ?

Reply:
No, because that's not same definition of mode.
- `transactions.file_open` is low-level and uses the following `mode` definition: `{os.O_RDONLY: "r", os.O_WRONLY: "w", os.O_RDWR: "rw"}`
- `workspace.open_file` is high level and uses the same definition as the `open` python built-in

I can add a comment explaining that.